cmd/crank/update.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/kong"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	typedclient "github.com/crossplane/crossplane/internal/client/clientset/versioned/typed/pkg/v1"
+
+	// Load all the auth plugins for the cloud providers.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+// updateCmd updates a package.
+type updateCmd struct {
+	Configuration updateConfigCmd   `cmd:"" help:"Update a Configuration package."`
+	Provider      updateProviderCmd `cmd:"" help:"Update a Provider package."`
+}
+
+// Run runs the update cmd.
+func (c *updateCmd) Run(b *buildChild) error {
+	return nil
+}
+
+// updateConfigCmd updates a Configuration.
+type updateConfigCmd struct {
+	Name string `arg:"" help:"Name of Configuration."`
+	Tag  string `arg:"" help:"Updated tag for Configuration package."`
+}
+
+// Run runs the Configuration update cmd.
+func (c *updateConfigCmd) Run(k *kong.Context) error { // nolint:gocyclo
+	kube := typedclient.NewForConfigOrDie(ctrl.GetConfigOrDie())
+	prevConf, err := kube.Configurations().Get(context.Background(), c.Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(warnIfNotFound(err), "cannot update configuration")
+	}
+	pkg := prevConf.Spec.Package
+	pkgReference, err := name.ParseReference(pkg)
+	if err != nil {
+		return errors.Wrap(warnIfNotFound(err), "cannot update configuration")
+	}
+	newPkg := ""
+	if strings.HasPrefix(c.Tag, "sha256") {
+		newPkg = pkgReference.Context().Digest(c.Tag).Name()
+	} else {
+		newPkg = pkgReference.Context().Tag(c.Tag).Name()
+	}
+	cr := &v1.Configuration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.Name,
+		},
+		Spec: v1.ConfigurationSpec{
+			PackageSpec: v1.PackageSpec{
+				Package: newPkg,
+			},
+		},
+	}
+	req, err := json.Marshal(cr)
+	if err != nil {
+		return errors.Wrap(warnIfNotFound(err), "cannot update configuration")
+	}
+	res, err := kube.Configurations().Patch(context.Background(), cr.Name, types.StrategicMergePatchType, req, metav1.PatchOptions{})
```

Comment:
@weastel I pulled down your fork here and tried this out and remembered that `StrategicMergePatch` is not supported for CRDs (ref https://github.com/kubernetes/kubernetes/issues/58414). We should be fine to use just `MergePatch` here.